### PR TITLE
Rest parameters and spread operator

### PR DESCRIPTION
**Pull request by luizbaldi**
Adding translation for "Rest parameters and spread operator".

Feel free to share thoughts on the translation and let me know if any changes are needed :)

### CODE REVIEW

**Comment by odsantos** [comment]
Hi, hope to do it as soon as possible. Thanks. Cheers.

File: 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md [hunk 1]
```diff
@@ -147,21 +148,21 @@ alert( Math.max(arr) ); // NaN
...
+Quando `...arr` é usado em uma chamada de função, ele "expande" um objeto iterável `arr` em uma lista de argumentos.
 
-For `Math.max`:
+Para `Math.max`:
 
 ```js run
 let arr = [3, 5, 1];
 
-alert( Math.max(...arr) ); // 5 (spread turns array into a list of arguments)
+alert( Math.max(...arr) ); // 5 (spread transforma uma matriz em uma lista de argumentos )
```
[the remark this answers is not in the document]
Comment (odsantos) [reply 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md]: There's a stray parenthesis there. Thanks.

**Comment by luizbaldi** [comment]
> look

Of course! Thanks for you review, gonna fix those changes asap :)

**Comment by odsantos** [comment]
Thanks.  
  On Fri, Oct 11, 2019 at 1:29 PM, Luiz Baldi<notifications@github.com> wrote:   

look


Of course! Thanks for you review, gonna fix those changes asap :)

**Comment by odsantos** [comment]
Hi, I hope to do it as soon as possible. Thanks.

File: 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md [hunk 2]
```diff
@@ -190,56 +189,55 @@ let arr2 = [8, 9, 15];
...
-- Rest parameters are used to create functions that accept any number of arguments.
-- The spread operator is used to pass an array to functions that normally require a list of many arguments.
+- Operadores _rest_ são usados para criar funções que aceitem qualquer número de argumentos.
+- O operador _spread_ é usado para passar um _array_ em funções que normalmente requerem uma lista de muitos argumentos.
 
-Together they help to travel between a list and an array of parameters with ease.
+Juntos, eles nos ajudam a efetuar a transição entre uma lista e um _array_ de parâmetros com facilidade.
 
-All arguments of a function call are also available in "old-style" `arguments`: array-like iterable object.
+Todos os argumentos de uma chamada de função também estão disponíveis na "moda antiga" `arguments`: objeto iterável tipo lista-genérica _(array-like)_.
```
Comment (odsantos) [inline 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md]: Please,

- '... objeto iterável do tipo lista-genérica (_array-like_).'
- Would you also move the emphasis  to inside the parentheses?
Thanks.

File: 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md [hunk 3]
```diff
@@ -190,56 +189,55 @@ let arr2 = [8, 9, 15];
...
+- `Array.from` opera tanto em listas-genéricas _(array-likes)_ como em iteráveis.
+- O operador spread opera somente em iteráveis.
 
+Então, para a tarefa de transformar algo em um _array_, `Array.from` tende a ser uma solução mais universal.
 
-## Summary
+## Sumário
 
-When we see `"..."` in the code, it is either rest parameters or the spread operator.
+Quando nos depararmos com `"..."` no código, estamos falando de parâmetros _rest_ ou operador _spread_.
```
Comment (odsantos) [inline 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md]: Please,

- '... de parâmetros _rest_ ou do operador _spread_.'

File: 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md [hunk 4]
```diff
@@ -190,56 +189,55 @@ let arr2 = [8, 9, 15];
...
-But there's a subtle difference between `Array.from(obj)` and `[...obj]`:
+O resultado é o mesmo que `...[str]`.
 
-- `Array.from` operates on both array-likes and iterables.
-- The spread operator operates only on iterables.
+Mas há uma diferença sútil entre `Array.from(obj)` e `[...obj]`:
 
-So, for the task of turning something into an array, `Array.from` tends to be more universal.
+- `Array.from` opera tanto em listas-genéricas _(array-likes)_ como em iteráveis.
+- O operador spread opera somente em iteráveis.
```
Comment (odsantos) [inline 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md]: Please,

Would you add emphasis to *spread*?
(Only to that, it is already good.)

File: 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md [hunk 5]
```diff
@@ -62,54 +62,54 @@ showName("Julius", "Caesar", "Consul", "Imperator");
...
+Antigamente, parâmetros _rest_ não existiam na linguagem, portanto usar `arguments` era a única forma de receber todos os argumentos da função, não importando o seu total de argumentos.
 
-And it still works, we can use it today.
+E isso ainda funciona atualmente.
 
-But the downside is that although `arguments` is both array-like and iterable, it's not an array. It does not support array methods, so we can't call `arguments.map(...)` for example.
+Mas a desvantagem é que, apesar do `arguments` ter algumas caracteristicas de _arrays_ e de ser iterável, não é um _array_ de fato. Ele não suporta métodos de _arrays_, por exemplo, não podemos chamar `arguments.map(...)`.
 
-Also, it always contains all arguments. We can't capture them partially, like we did with rest parameters.
+E além disso, sempre contém todos os argumentos. Não podemos os obter parcialmente, como fizemos com parâmetros _rest_.
```
[the remark this answers is not in the document]
Comment (luizbaldi) [reply 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md]: ?

File: 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md [hunk 6]
```diff
@@ -190,56 +189,55 @@ let arr2 = [8, 9, 15];
...
-- Rest parameters are used to create functions that accept any number of arguments.
-- The spread operator is used to pass an array to functions that normally require a list of many arguments.
+- Operadores _rest_ são usados para criar funções que aceitem qualquer número de argumentos.
+- O operador _spread_ é usado para passar um _array_ em funções que normalmente requerem uma lista de muitos argumentos.
 
-Together they help to travel between a list and an array of parameters with ease.
+Juntos, eles nos ajudam a efetuar a transição entre uma lista e um _array_ de parâmetros com facilidade.
 
-All arguments of a function call are also available in "old-style" `arguments`: array-like iterable object.
+Todos os argumentos de uma chamada de função também estão disponíveis na "moda antiga" `arguments`: objeto iterável tipo lista-genérica (_array-like_).
```
Comment (odsantos) [inline 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md]: Please,

- '... objeto iterável do tipo lista-genérica (_array-like_).'

File: 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md [hunk 7]
```diff
@@ -121,23 +121,23 @@ f(1); // 1
 ```
 ````
```
Comment (odsantos) [inline 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md]: Please,

- The code above seems to contain an error conflict. 
So, in such situations would you check the [English article](https://javascript.info/rest-parameters-spread-operator) (there is a link here) for ways to resolve the conflict?
From the text there, you might know how the Portuguese page would look.

File: 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md [hunk 8]
```diff
@@ -147,21 +147,21 @@ alert( Math.max(arr) ); // NaN
 */!*
 ```
```
Comment (odsantos) [inline 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md]: Please,

- The code above seems to contain an error conflict.
(In doubt, would you check the English website [article](https://javascript.info/rest-parameters-spread-operator) (link here)?)

File: 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md [hunk 9]
```diff
@@ -170,8 +170,7 @@ let arr2 = [8, 3, -8, 1];
 alert( Math.max(...arr1, ...arr2) ); // 8
 ```
```
Comment (odsantos) [inline 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md]: Please,

- The code above seems to contain an error conflict.
(Would you check the English website article (link above)?)

File: 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md [hunk 10]
```diff
@@ -180,7 +179,7 @@ let arr2 = [8, 3, -8, 1];
 alert( Math.max(1, ...arr1, 2, ...arr2, 25) ); // 25
 ```
```
[the remark this answers is not in the document]
Comment (luizbaldi) [reply 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md]: I don't what you mean by "error conflict". The line 179 looks exactly like the original one on the english version, as we can see [here](https://github.com/javascript-tutorial/en.javascript.info/blob/master/1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md#L179)
]. Can you explain to me what exactly should I do here?

File: 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md [hunk 11]
```diff
@@ -190,56 +189,55 @@ let arr2 = [8, 9, 15];
 let merged = [0, ...arr, 2, ...arr2];
 */!*
 
-alert(merged); // 0,3,5,1,2,8,9,15 (0, then arr, then 2, then arr2)
+alert(merged); // 0,3,5,1,2,8,9,15 (0, depois arr, depois 2, depois arr2)
 ```
```
[the remark this answers is not in the document]
Comment (luizbaldi) [reply 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md]: I don't what you mean by "error conflict". The line 179 looks exactly like the original one on the english version, as we can see [here](https://github.com/javascript-tutorial/en.javascript.info/blob/master/1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md#L192). Can you explain to me what exactly should I do here?

File: 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md (hunk 7 above)
Comment (luizbaldi) [reply 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md]: I didn't get this "error conflicts" at all, sorry. Can you explain me better what you mean by that?

File: 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md (hunk 8 above)
Comment (luizbaldi) [reply 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md]: Same goes here.

File: 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md (hunk 9 above)
Comment (luizbaldi) [reply 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md]: Same.

File: 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (odsantos) [reply 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md]: - Please, check [Spread operator](https://javascript.info/rest-parameters-spread-operator#spread-operator) - 6th code block:
**The second array (arr2) is missing**
**And also formatting characters**

File: 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md [hunk 12]
```diff
@@ -190,56 +189,55 @@ let arr2 = [8, 9, 15];
...
 
-alert( [...str] ); // H,e,l,l,o
+alert( [...str] ); // O,l,á
 ```
 
-The spread operator internally uses iterators to gather elements, the same way as `for..of` does.
+Internamente, o operador _spread_ usa iteradores (_iterators_) para reunir elementos, da mesma forma como `for..of` faz.
 
-So, for a string, `for..of` returns characters and `...str` becomes `"H","e","l","l","o"`. The list of characters is passed to array initializer `[...str]`.
+Então, para uma string, `for..of` retorna caracteres, e `...str` se torna `"O","l","á"`. A lista de caracteres é passada para o inicializador do _array_ `[...str]`
```
Comment (odsantos) [inline 1-js/06-advanced-functions/02-rest-parameters-spread-operator/article.md]: Please,
- "... retorna caracteres, e `...str` se torna **em** `"O","l","á"`."
(but do **not** format in bold)

**Pull request closed without merging**